text/0715-construct-upgrade-support.md

```diff
@@ -0,0 +1,786 @@
+code changes that could trigger CloudFormation resource replacements (e.g., NAT gateways being
+recreated, DynamoDB global tables being recreated). Recreation of resources leads to data loss for
+stateful resources, and can lead to loss of availability for others.
+
+To address this, this construct upgrade guide demonstrates the workflow for safely upgrading constructs
+through guided documentation, validation commands, and deployment commands. This ensures users can execute
+construct upgrades in a controlled and safe manner.
+
+## Working Backward
+
+When upgrading CDK constructs, CDK users follow these steps using our construct upgrade guides and CDK CLI
+commands:
+
+- **Review Migration Guide:** Developers start by reading CDK-provided comprehensive migration guide for
+their specific construct upgrade (e.g., DynamoDB Table to TableV2). The details of individual migration
+guide is out of scope of this RFC.
+- **Update CDK Constructs:** Developers update their CDK stack to V2 variants of the constructs following the
+construct upgrade guide.
+- **Validate Changes:** Developers run CDK CLI commands to validate their code changes.
+- **Execute Changes:** Developers run the corresponding execute command in CDK CLI to deploy the changes.
+
+### Working Backward: DynamoDB Table Upgrade Example
+
+Below is a common upgrade scenario to demonstrate how the CDK Construct Upgrade framework helps developers
+safely migrate their infrastructure. Consider a DynamoDB table deployed to CloudFormation using the legacy
+`Table` construct:
+
+```ts
+import * as dynamodb from 'aws-cdk-lib/aws-dynamodb';
+
+export class DemoStack extends cdk.Stack {
+  constructor(scope: Construct, id: string, props?: cdk.StackProps) {
+    super(scope, id, props);
+
+    const table = new dynamodb.Table(this, 'MyTable', {
+      partitionKey: {
+        name: 'PK',
+        type: dynamodb.AttributeType.STRING,
+      },
+      replicationRegions: ['us-west-2'],
+    });
+  }
+}
+```
+
+**Step 1:**: CDK users follow the construct upgrade guide provided by CDK team and upgrade to TableV2 construct:
+
+```ts
+import * as dynamodb from 'aws-cdk-lib/aws-dynamodb';
+
+export class DemoStack extends cdk.Stack {
+  constructor(scope: Construct, id: string, props?: cdk.StackProps) {
+    super(scope, id, props);
+
+    const table = new dynamodb.TableV2(this, 'MyTable', {
+      partitionKey: {
+        name: 'PK',
+        type: dynamodb.AttributeType.STRING,
+      },
+      tableName: 'DemoStack-MyTable794EDED1-11W4MR8VZ0UPE',
+      replicas: [
+        {
+          region: 'us-west-2',
+        }
+      ]
+    });
+  }
+}
+```
+
+**Step 2:**: CDK users follow the construct upgrade guide to validate code changes by running the
+CDK CLI commands below:
+
+```s
+# exit with 1 if CloudFormation drift is detected for resource type AWS::DynamoDB::Table
+# exit with 0 otherwise
+$ cdk drift --rejectAWS::DynamoDB::Table
```

Comment:
Why do we need to reject specifically this? Would we not reject on any drift at all?

(I mean, it's fine to have options on the command if a user knows they want to ignore drift... but it shouldn't [need to be] be part of the **expected** flow)